### PR TITLE
Add seeded dropout

### PR DESCRIPTION
**Pull request by Dref360**
## Summary:

Add a way to set the seed for Dropout so that the execution is always the same.

### Features:

* Add SeededDropout

## Checklist:

* [x] Your code is documented (To validate this, add your module to `tests/documentation_test.py`).
* [x] Your code is tested with unit tests.
* [x] You moved your Issue to the PR state.

### CODE REVIEW

File: src/baal/bayesian/seeded_dropout.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+            self.cpu()
+
+    def cuda(self, device=None):
+        if not self.use_cuda or self.gen is None:
+            self.use_cuda = True
+            self.gen = torch.Generator('cuda')
+            self.gen.manual_seed(self.seed)
+        return super().cuda(device)
+
+    def cpu(self, device=None):
```
Comment (Dref360) [inline src/baal/bayesian/seeded_dropout.py]:
```suggestion
    def cpu(self):
```

**Comment by janfreyberg** [comment]
LGTM - for coverage, we could add lines that include is_cuda / use_cuda / similar keywords in the `.coveragerc` file?

**Comment by BlackHC** [comment]
I'm a bit wary of using fixed seeds as a way of ensuring consistency. This works well for simple models but could fail when the batch size changes during inference (usually the last batch might be incomplete and drop_last=False by default in pytorch's dataloader).

Moreover, 3x slower might also be an issue.

**Comment by Dref360** [comment]
@BlackHC I agree that this is far from perfect. And even in a research setup, the 3x slowdown would be problematic.

How should BaaL ensure consistency? (Open questions to everyone)

**Comment by BlackHC** [comment]
In our code, we fixed the masks (which means you also have batching along the MC dropout samples dimension) and we had multiple iterations over the input data.

**Comment by Dref360** [comment]
Will close in favor of Andreas proposal :)
